crates/proto/src/rr/rdata/cert.rs

```diff
@@ -0,0 +1,762 @@
+
+    ///   18-22, 24-122    Unassigned
+    Unassigned(u8),
+}
+
+impl From<u8> for Algorithm {
+    fn from(algorithm: u8) -> Self {
+        match algorithm {
+            0 => Self::Reserved(0),
+            1 => Self::RSAMD5,
+            2 => Self::DH,
+            3 => Self::DSA,
+            4 => Self::ECC,
+            5 => Self::RSASHA1,
+            6 => Self::DSANSEC3SHA1,
+            7 => Self::RSASHA1NSEC3SHA1,
+            8 => Self::RSASHA256,
+            9 => Self::Reserved(9),
+            10 => Self::RSASHA512,
+            11 => Self::Reserved(11),
+            12 => Self::ECCGOST,
+            13 => Self::ECDSAP256SHA256,
+            14 => Self::ECDSAP384SHA384,
+            15 => Self::ED25519,
+            16 => Self::ED448,
+            17 => Self::SM2SM3,
+            18..=22 => Self::Unassigned(algorithm),
+            23 => Self::ECCGOST12,
+            24..=122 => Self::Unassigned(algorithm),
+            252 => Self::INDIRECT,
+            253 => Self::PRIVATEDNS,
+            254 => Self::PRIVATEOID,
+            _ => Self::Unassigned(algorithm),
+        }
+    }
+}
+
+impl From<Algorithm> for u8 {
+    fn from(algorithm: Algorithm) -> Self {
+        match algorithm {
+            Algorithm::Reserved(value) if value == 0 => value,
+            Algorithm::RSAMD5 => 1,
+            Algorithm::DH => 2,
+            Algorithm::DSA => 3,
+            Algorithm::ECC => 4,
+            Algorithm::RSASHA1 => 5,
+            Algorithm::DSANSEC3SHA1 => 6,
+            Algorithm::RSASHA1NSEC3SHA1 => 7,
+            Algorithm::RSASHA256 => 8,
+            Algorithm::Reserved(value) if value == 9 => value,
+            Algorithm::RSASHA512 => 10,
+            Algorithm::Reserved(value) if value == 11 => value,
+            Algorithm::ECCGOST => 12,
+            Algorithm::ECDSAP256SHA256 => 13,
+            Algorithm::ECDSAP384SHA384 => 14,
+            Algorithm::ED25519 => 15,
+            Algorithm::ED448 => 16,
+            Algorithm::SM2SM3 => 17,
+            Algorithm::Unassigned(value) if (18..=22).contains(&value) => value,
+            Algorithm::ECCGOST12 => 23,
+            Algorithm::Unassigned(value) if (24..=122).contains(&value) => value,
+            Algorithm::INDIRECT => 252,
+            Algorithm::PRIVATEDNS => 253,
+            Algorithm::PRIVATEOID => 254,
+            Algorithm::Unassigned(value) => value,
+            Algorithm::Reserved(value) => value,
+        }
+    }
+}
+
+impl<'r> BinDecodable<'r> for Algorithm {
+    // https://www.iana.org/assignments/dns-sec-alg-numbers/dns-sec-alg-numbers.xhtml
+    fn read(decoder: &mut BinDecoder<'r>) -> ProtoResult<Self> {
+        let algorithm_id = decoder
+            .read_u8()?
+            .unverified(/*Algorithm is verified as safe in processing this*/);
+        Ok(Self::from(algorithm_id))
+    }
+}
+
+impl fmt::Display for Algorithm {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
```

Comment:
I do realize I'm (very) late to the party, but why was this added? This is the same data type as https://github.com/hickory-dns/hickory-dns/blob/7887002d1198ed9faaaae531e0f9815c10589f7a/crates/proto/src/rr/dnssec/algorithm.rs#L105?